### PR TITLE
Erc721 support

### PR DESCRIPTION
**Pull request by eberger727**
Support for ERC721 tokens

### CODE REVIEW

File: solidity/contracts/ERC721WithData.sol [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ) external {
+        safeTransferFrom(from, to, tokenId, data);
+    }
+
+    function burnWithData(
+        address from,
+        uint256 tokenId,
+        bytes calldata data
+    ) external {
+        require(from == _msgSender(), 'ERC20WithData: caller is not owner');
```
Comment (awrichar) [inline solidity/contracts/ERC721WithData.sol:37]: Wrong name in error message

File: src/tokens/tokens.service.ts [hunk 2]
```diff
@@ -100,19 +108,34 @@ export class TokensService {
...
+  private getAmountOrTokenID(
+    dto: TokenMint | TokenTransfer | TokenBurn,
+    type: TokenType,
+  ): string | undefined {
+    if (type === TokenType.FUNGIBLE) {
+      return dto.amount;
+    }
+
+    if (dto.amount !== undefined && dto.amount !== '1') {
+      throw new HttpException('Amount for nonfungible tokens must be 1', HttpStatus.BAD_REQUEST);
```
Comment (awrichar) [inline src/tokens/tokens.service.ts:133]: You can use `BadRequestException` as a shorthand (amounts to the same thing though)

File: src/tokens/tokens.service.ts [hunk 3]
```diff
@@ -370,4 +437,20 @@ class TokenListener implements EventListener {
...
+  private postOptions(operator: string) {
+    const from = `${this.service.shortPrefix}-from`;
+    const sync = `${this.service.shortPrefix}-sync`;
+    const call = `${this.service.shortPrefix}-call`;
+
+    const requestOptions: AxiosRequestConfig = {
+      params: {
+        [from]: operator,
+        [sync]: 'false',
+        [call]: 'true',
```
Comment (awrichar) [inline src/tokens/tokens.service.ts:449]: I know you only have one usage right now and it is a `call`, but it seems like maybe this should be passed in as a bool for readability/flexibility.

**Review by awrichar — APPROVED** [review]
Couple of minor comments if you want to address them - but marking approved.

File: src/tokens/tokens.service.ts (hunk 3 above)
Comment (eberger727) [reply src/tokens/tokens.service.ts:449]: done

**Pull request merged**